server/src/main/java/org/cloudfoundry/identity/uaa/provider/IdentityProviderEndpoints.java

```diff
@@ -308,6 +309,23 @@ public ResponseEntity<IdentityProviderStatus> updateIdentityProviderStatus(@Path
         return  new ResponseEntity<>(body, OK);
     }
 
+    @DeleteMapping(value = "{id}/secret")
+    public ResponseEntity<IdentityProvider> deleteSecret(@PathVariable String id) {
+        String zoneId = identityZoneManager.getCurrentIdentityZoneId();
+        IdentityProvider existing = identityProviderProvisioning.retrieve(id, zoneId);
+        if((OIDC10.equals(existing.getType()) || OAUTH20.equals(existing.getType()))
+            && existing.getConfig() instanceof AbstractExternalOAuthIdentityProviderDefinition<?> idpConfiguration) {
+            idpConfiguration.setRelyingPartySecret(null);
+            identityProviderProvisioning.update(existing, zoneId);
```

Comment:
We also need to ensure that this update is performed on any potential alias of the IdP

Reply:
ok